src/internal/connector/exchange/api/mail_test.go

```diff
@@ -145,3 +153,202 @@ func (suite *MailAPIUnitSuite) TestMailInfo() {
 		})
 	}
 }
+
+type MailAPIE2ESuite struct {
+	tester.Suite
+	credentials account.M365Config
+	ac          Client
+	user        string
+}
+
+// We do end up mocking the actual request, but creating the rest
+// similar to E2E suite
+func TestMailAPIE2ESuite(t *testing.T) {
+	suite.Run(t, &MailAPIE2ESuite{
+		Suite: tester.NewIntegrationSuite(
+			t,
+			[][]string{tester.M365AcctCredEnvs},
+		),
+	})
+}
+
+func (suite *MailAPIE2ESuite) SetupSuite() {
+	t := suite.T()
+
+	a := tester.NewM365Account(t)
+	m365, err := a.M365Config()
+	require.NoError(t, err, clues.ToCore(err))
+
+	suite.credentials = m365
+	suite.ac, err = NewMockableClient(m365)
+	require.NoError(t, err, clues.ToCore(err))
+
+	suite.user = tester.M365UserID(suite.T())
+}
+
+func getJSONObject(t *testing.T, thing serialization.Parsable) map[string]interface{} {
+	sw := kjson.NewJsonSerializationWriter()
+
+	err := sw.WriteObjectValue("", thing)
+	require.NoError(t, err, "serialize")
+
+	content, err := sw.GetSerializedContent()
+	require.NoError(t, err, "serialize")
+
+	var out map[string]interface{}
+	err = json.Unmarshal([]byte(content), &out)
+	require.NoError(t, err, "unmarshall")
+
+	return out
+}
+
+func (suite *MailAPIE2ESuite) TestHugeAttachmentListDownload() {
+	mid := "fake-message-id"
+	aid := "fake-attachment-id"
+
+	tests := []struct {
+		name            string
+		setupf          func()
+		attachmentCount int
+		expect          assert.ErrorAssertionFunc
+	}{
+		{
+			name: "simple fetch",
```

Comment:
nit: just to be a little more descriptive
```suggestion
			name: "no attachments",
```